kcauto.sikuli/pvp.sikuli/pvp.py

```diff
@@ -73,40 +73,25 @@ def run_pvp_logic(self):
             opponent_counts['ss'] + opponent_counts['ssv']))
         Util.log_msg("Selecting {} and {!s} on night battle.".format(
             formation.replace('_', ' '), night_battle))
-
         # start pvp
         Util.wait_and_click(self.regions['lower_left'], 'pvp_start_1.png', 30)
         Util.wait_and_click(self.regions['lower'], 'pvp_start_2.png', 30)
         Util.log_msg("Beginning PvP sortie.")
         Util.rejigger_mouse(self.regions, 'top')
         Util.wait_and_click(self.regions[formation], formation)
 
-        while not (
-                self.regions['lower_right_corner'].exists('next.png') or
-                self.kc_region.exists('combat_nb_fight.png')):
-            # wait through combat
-            pass
-
-        # resolve night battle
-        if self.kc_region.exists('combat_nb_fight.png'):
-            if (night_battle):
-                Util.check_and_click(
-                    self.kc_region, 'combat_nb_fight.png')
-            else:
-                Util.check_and_click(
-                    self.kc_region, 'combat_nb_retreat.png')
-
-        while not self.regions['lower_right_corner'].exists('next.png'):
-            # wait through night battle combat, if applicable
-            pass
-
-        Util.click_preset_region(self.regions, 'center')
-
-        while not self.regions['home_menu'].exists('home_menu_sortie.png'):
-            # click through post-combat screens until main menu
-            Util.click_preset_region(self.regions, 'shipgirl')
-            Util.kc_sleep(2)
-
+        while not self.regions['lower_right_corner'].exists('next.png', 1 + Globals.SLEEP_MODIFIER):
+            if self.kc_region.exists('combat_nb_fight.png', 1 + Globals.SLEEP_MODIFIER):
+                if night_battle:
+                    Util.check_and_click(self.kc_region, 'combat_nb_fight.png')
+                else:
+                    Util.check_and_click(self.kc_region, 'combat_nb_retreat.png')
+                Util.rejigger_mouse(self.regions, 'top')
+
+        while not self.regions['home_menu'].exists('home_menu_sortie.png', 1 + Globals.SLEEP_MODIFIER):
+            if self.regions['lower_right_corner'].exists('next.png', 1 + Globals.SLEEP_MODIFIER):
+                Util.check_and_click(self.regions['lower_right_corner'], 'next.png')
```

Comment:
I personally think that this is an OK place to use `click_preset_region`...

Reply:
If you are going to persist on moving away from `click_preset_region` here (it doesn't really matter to me), please set up a new `EXPAND` keyword in `kca_globals` so that it clicks the center of the screen or the shipgirl area relative to the button. One of the purpose of `click_preset_region` was to provide a convenient way to define a larger click region for anti-anti-bot measures, and I'd like to not lose that.

Reply:
Sure, `next.png` is small enough.  I'll implement this later